{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I went to following link: https://evolgeniusteam.github.io/gmrepodocumentation/usage/downloaddatafromgmrepo/\n",
    "\n",
    "I download Processed runs (samples_loaded.txt here) that has both uid and sample id which help to map samples to relative abundance profile\n",
    "\n",
    "I downloaded relative abundances (species_abundance.txt here) that has relative abundance for each taxon for eaxh sample. I made relative abundance profile based on it\n",
    "\n",
    "I downloaded All runs (samlpe_to_run_info.txt here) that has metadata for each sample. \n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### whole data preprocessing and filtering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Set the display option to show all rows\n",
    "# pd.set_option('display.max_rows', None)\n",
    "# pd.set_option('display.min_rows', None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "relative_abundance = pd.read_csv('species_abundance.txt', delimiter='\\t')  # Adjust the delimiter as needed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(8283371, 4)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relative_abundance.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['loaded_uid', 'ncbi_taxon_id', 'taxon_rank_level',\n",
      "       'relative_abundance'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "print(relative_abundance.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "relative_abundance_df = relative_abundance.pivot_table(index='loaded_uid', columns='ncbi_taxon_id', values='relative_abundance', fill_value=0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ncbi_taxon_id</th>\n",
       "      <th>-1</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>...</th>\n",
       "      <th>2212691</th>\n",
       "      <th>2212731</th>\n",
       "      <th>2282523</th>\n",
       "      <th>2282740</th>\n",
       "      <th>2282741</th>\n",
       "      <th>2282742</th>\n",
       "      <th>2304691</th>\n",
       "      <th>2304692</th>\n",
       "      <th>2339232</th>\n",
       "      <th>2529408</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>loaded_uid</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.503245</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11.210800</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.496660</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>5.832720</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>5.147945</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52855</th>\n",
       "      <td>14.686035</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52856</th>\n",
       "      <td>2.268080</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52857</th>\n",
       "      <td>3.817365</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52858</th>\n",
       "      <td>46.556050</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52859</th>\n",
       "      <td>5.332600</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32244 rows × 8910 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "ncbi_taxon_id   -1         6         7         9         10        11       \\\n",
       "loaded_uid                                                                   \n",
       "1               1.503245       0.0       0.0       0.0       0.0       0.0   \n",
       "2              11.210800       0.0       0.0       0.0       0.0       0.0   \n",
       "3               0.496660       0.0       0.0       0.0       0.0       0.0   \n",
       "6               5.832720       0.0       0.0       0.0       0.0       0.0   \n",
       "7               5.147945       0.0       0.0       0.0       0.0       0.0   \n",
       "...                  ...       ...       ...       ...       ...       ...   \n",
       "52855          14.686035       0.0       0.0       0.0       0.0       0.0   \n",
       "52856           2.268080       0.0       0.0       0.0       0.0       0.0   \n",
       "52857           3.817365       0.0       0.0       0.0       0.0       0.0   \n",
       "52858          46.556050       0.0       0.0       0.0       0.0       0.0   \n",
       "52859           5.332600       0.0       0.0       0.0       0.0       0.0   \n",
       "\n",
       "ncbi_taxon_id   13        14        16        17       ...   2212691  \\\n",
       "loaded_uid                                             ...             \n",
       "1                   0.0       0.0       0.0       0.0  ...       0.0   \n",
       "2                   0.0       0.0       0.0       0.0  ...       0.0   \n",
       "3                   0.0       0.0       0.0       0.0  ...       0.0   \n",
       "6                   0.0       0.0       0.0       0.0  ...       0.0   \n",
       "7                   0.0       0.0       0.0       0.0  ...       0.0   \n",
       "...                 ...       ...       ...       ...  ...       ...   \n",
       "52855               0.0       0.0       0.0       0.0  ...       0.0   \n",
       "52856               0.0       0.0       0.0       0.0  ...       0.0   \n",
       "52857               0.0       0.0       0.0       0.0  ...       0.0   \n",
       "52858               0.0       0.0       0.0       0.0  ...       0.0   \n",
       "52859               0.0       0.0       0.0       0.0  ...       0.0   \n",
       "\n",
       "ncbi_taxon_id   2212731   2282523   2282740   2282741   2282742   2304691  \\\n",
       "loaded_uid                                                                  \n",
       "1                   0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "2                   0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "3                   0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "6                   0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "7                   0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "...                 ...       ...       ...       ...       ...       ...   \n",
       "52855               0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "52856               0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "52857               0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "52858               0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "52859               0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "\n",
       "ncbi_taxon_id   2304692   2339232   2529408  \n",
       "loaded_uid                                   \n",
       "1                   0.0       0.0       0.0  \n",
       "2                   0.0       0.0       0.0  \n",
       "3                   0.0       0.0       0.0  \n",
       "6                   0.0       0.0       0.0  \n",
       "7                   0.0       0.0       0.0  \n",
       "...                 ...       ...       ...  \n",
       "52855               0.0       0.0       0.0  \n",
       "52856               0.0       0.0       0.0  \n",
       "52857               0.0       0.0       0.0  \n",
       "52858               0.0       0.0       0.0  \n",
       "52859               0.0       0.0       0.0  \n",
       "\n",
       "[32244 rows x 8910 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relative_abundance_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Drop the column named '-1'\n",
    "relative_abundance_df = relative_abundance_df.drop(columns=[-1])\n",
    "\n",
    "# Remove columns that have less than 2 non-zero values\n",
    "relative_abundance_df = relative_abundance_df.loc[:, (relative_abundance_df != 0).sum(axis=0) >= 2]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ncbi_taxon_id</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>18</th>\n",
       "      <th>...</th>\n",
       "      <th>2211641</th>\n",
       "      <th>2212691</th>\n",
       "      <th>2212731</th>\n",
       "      <th>2282523</th>\n",
       "      <th>2282740</th>\n",
       "      <th>2282741</th>\n",
       "      <th>2282742</th>\n",
       "      <th>2304691</th>\n",
       "      <th>2304692</th>\n",
       "      <th>2529408</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>loaded_uid</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52855</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52856</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52857</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52858</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52859</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32244 rows × 8599 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "ncbi_taxon_id  6        7        9        10       11       13       14       \\\n",
       "loaded_uid                                                                     \n",
       "1                  0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "2                  0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "3                  0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "6                  0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "7                  0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "...                ...      ...      ...      ...      ...      ...      ...   \n",
       "52855              0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "52856              0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "52857              0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "52858              0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "52859              0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "\n",
       "ncbi_taxon_id  16       17       18       ...  2211641  2212691  2212731  \\\n",
       "loaded_uid                                ...                              \n",
       "1                  0.0      0.0      0.0  ...      0.0      0.0      0.0   \n",
       "2                  0.0      0.0      0.0  ...      0.0      0.0      0.0   \n",
       "3                  0.0      0.0      0.0  ...      0.0      0.0      0.0   \n",
       "6                  0.0      0.0      0.0  ...      0.0      0.0      0.0   \n",
       "7                  0.0      0.0      0.0  ...      0.0      0.0      0.0   \n",
       "...                ...      ...      ...  ...      ...      ...      ...   \n",
       "52855              0.0      0.0      0.0  ...      0.0      0.0      0.0   \n",
       "52856              0.0      0.0      0.0  ...      0.0      0.0      0.0   \n",
       "52857              0.0      0.0      0.0  ...      0.0      0.0      0.0   \n",
       "52858              0.0      0.0      0.0  ...      0.0      0.0      0.0   \n",
       "52859              0.0      0.0      0.0  ...      0.0      0.0      0.0   \n",
       "\n",
       "ncbi_taxon_id  2282523  2282740  2282741  2282742  2304691  2304692  2529408  \n",
       "loaded_uid                                                                    \n",
       "1                  0.0      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "2                  0.0      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "3                  0.0      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "6                  0.0      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "7                  0.0      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "...                ...      ...      ...      ...      ...      ...      ...  \n",
       "52855              0.0      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "52856              0.0      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "52857              0.0      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "52858              0.0      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "52859              0.0      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "\n",
       "[32244 rows x 8599 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relative_abundance_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "relative_abundance_df.to_csv('relative_abundance.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "relative_abundance_df = pd.read_csv('relative_abundance.csv', index_col = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>loaded_uid</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>...</th>\n",
       "      <th>2211641</th>\n",
       "      <th>2212691</th>\n",
       "      <th>2212731</th>\n",
       "      <th>2282523</th>\n",
       "      <th>2282740</th>\n",
       "      <th>2282741</th>\n",
       "      <th>2282742</th>\n",
       "      <th>2304691</th>\n",
       "      <th>2304692</th>\n",
       "      <th>2529408</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32239</th>\n",
       "      <td>52855</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32240</th>\n",
       "      <td>52856</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32241</th>\n",
       "      <td>52857</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32242</th>\n",
       "      <td>52858</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32243</th>\n",
       "      <td>52859</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32244 rows × 8600 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       loaded_uid    6    7    9   10   11   13   14   16   17  ...  2211641  \\\n",
       "0               1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "1               2  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "2               3  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "3               6  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "4               7  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "...           ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...      ...   \n",
       "32239       52855  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "32240       52856  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "32241       52857  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "32242       52858  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "32243       52859  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "\n",
       "       2212691  2212731  2282523  2282740  2282741  2282742  2304691  2304692  \\\n",
       "0          0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "1          0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "2          0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "3          0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "4          0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "...        ...      ...      ...      ...      ...      ...      ...      ...   \n",
       "32239      0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "32240      0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "32241      0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "32242      0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "32243      0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "\n",
       "       2529408  \n",
       "0          0.0  \n",
       "1          0.0  \n",
       "2          0.0  \n",
       "3          0.0  \n",
       "4          0.0  \n",
       "...        ...  \n",
       "32239      0.0  \n",
       "32240      0.0  \n",
       "32241      0.0  \n",
       "32242      0.0  \n",
       "32243      0.0  \n",
       "\n",
       "[32244 rows x 8600 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relative_abundance_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3045288/477576356.py:1: DtypeWarning: Columns (0,2,5,14,15,22,26,27,28,29) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  metadata = pd.read_csv('sample_to_run_info.txt', delimiter='\\t')\n"
     ]
    }
   ],
   "source": [
    "metadata = pd.read_csv('sample_to_run_info.txt', delimiter='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>checking</th>\n",
       "      <th>project_id</th>\n",
       "      <th>our_project_id</th>\n",
       "      <th>sample_name</th>\n",
       "      <th>original_sample_description</th>\n",
       "      <th>curated_sample_description</th>\n",
       "      <th>run_id</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>second_sample_id</th>\n",
       "      <th>experiment_type</th>\n",
       "      <th>...</th>\n",
       "      <th>sex</th>\n",
       "      <th>host_age</th>\n",
       "      <th>diet</th>\n",
       "      <th>longitude</th>\n",
       "      <th>latitude</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Recent.Antibiotics.Use</th>\n",
       "      <th>antibiotics_used</th>\n",
       "      <th>Antibiotics.Dose</th>\n",
       "      <th>Days.Without.Antibiotics.Use</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJDB4360</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Japanese001</td>\n",
       "      <td>Healthy Japanese gut microbiota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DRR048993</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD00042843</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>28.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>139.680</td>\n",
       "      <td>35.4800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJDB4360</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Japanese002</td>\n",
       "      <td>Healthy Japanese gut microbiota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DRR048994</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD00042844</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>Male</td>\n",
       "      <td>31.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>139.680</td>\n",
       "      <td>35.4800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJDB4360</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Japanese003</td>\n",
       "      <td>Healthy Japanese gut microbiota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DRR048995</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD00042845</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>Male</td>\n",
       "      <td>60.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>139.680</td>\n",
       "      <td>35.4800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJDB4360</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Japanese004</td>\n",
       "      <td>Healthy Japanese gut microbiota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DRR048996</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD00042846</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>Male</td>\n",
       "      <td>57.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>139.680</td>\n",
       "      <td>35.4800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJDB4360</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Japanese005</td>\n",
       "      <td>Healthy Japanese gut microbiota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DRR048997</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD00042847</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>Male</td>\n",
       "      <td>29.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>139.680</td>\n",
       "      <td>35.4800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66575</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA215106</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F34_CY</td>\n",
       "      <td>Functional metagenomic selections were perform...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR961853</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>Metagenomics</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-122.268</td>\n",
       "      <td>47.6614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66576</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA215106</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F34_PE</td>\n",
       "      <td>Functional metagenomic selections were perform...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR961854</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>Metagenomics</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-122.268</td>\n",
       "      <td>47.6614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66577</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA215106</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F34_TE</td>\n",
       "      <td>Functional metagenomic selections were perform...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR961855</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>Metagenomics</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-122.268</td>\n",
       "      <td>47.6614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66578</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA215106</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F34_TR</td>\n",
       "      <td>Functional metagenomic selections were perform...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR961856</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>Metagenomics</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-122.268</td>\n",
       "      <td>47.6614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66579</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA215106</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F34_TRSX</td>\n",
       "      <td>Functional metagenomic selections were perform...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR961857</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>Metagenomics</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-122.268</td>\n",
       "      <td>47.6614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>66580 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      checking   project_id our_project_id  sample_name  \\\n",
       "0            1    PRJDB4360            NaN  Japanese001   \n",
       "1            1    PRJDB4360            NaN  Japanese002   \n",
       "2            1    PRJDB4360            NaN  Japanese003   \n",
       "3            1    PRJDB4360            NaN  Japanese004   \n",
       "4            1    PRJDB4360            NaN  Japanese005   \n",
       "...        ...          ...            ...          ...   \n",
       "66575        1  PRJNA215106            NaN       F34_CY   \n",
       "66576        1  PRJNA215106            NaN       F34_PE   \n",
       "66577        1  PRJNA215106            NaN       F34_TE   \n",
       "66578        1  PRJNA215106            NaN       F34_TR   \n",
       "66579        1  PRJNA215106            NaN     F34_TRSX   \n",
       "\n",
       "                             original_sample_description  \\\n",
       "0                        Healthy Japanese gut microbiota   \n",
       "1                        Healthy Japanese gut microbiota   \n",
       "2                        Healthy Japanese gut microbiota   \n",
       "3                        Healthy Japanese gut microbiota   \n",
       "4                        Healthy Japanese gut microbiota   \n",
       "...                                                  ...   \n",
       "66575  Functional metagenomic selections were perform...   \n",
       "66576  Functional metagenomic selections were perform...   \n",
       "66577  Functional metagenomic selections were perform...   \n",
       "66578  Functional metagenomic selections were perform...   \n",
       "66579  Functional metagenomic selections were perform...   \n",
       "\n",
       "      curated_sample_description     run_id     sample_id second_sample_id  \\\n",
       "0                            NaN  DRR048993           NaN     SAMD00042843   \n",
       "1                            NaN  DRR048994           NaN     SAMD00042844   \n",
       "2                            NaN  DRR048995           NaN     SAMD00042845   \n",
       "3                            NaN  DRR048996           NaN     SAMD00042846   \n",
       "4                            NaN  DRR048997           NaN     SAMD00042847   \n",
       "...                          ...        ...           ...              ...   \n",
       "66575                        NaN  SRR961853  SAMN02317178     SAMN02317178   \n",
       "66576                        NaN  SRR961854  SAMN02317178     SAMN02317178   \n",
       "66577                        NaN  SRR961855  SAMN02317178     SAMN02317178   \n",
       "66578                        NaN  SRR961856  SAMN02317178     SAMN02317178   \n",
       "66579                        NaN  SRR961857  SAMN02317178     SAMN02317178   \n",
       "\n",
       "      experiment_type  ...     sex host_age diet longitude latitude BMI  \\\n",
       "0            Amplicon  ...  Female     28.0  NaN   139.680  35.4800 NaN   \n",
       "1            Amplicon  ...    Male     31.0  NaN   139.680  35.4800 NaN   \n",
       "2            Amplicon  ...    Male     60.0  NaN   139.680  35.4800 NaN   \n",
       "3            Amplicon  ...    Male     57.0  NaN   139.680  35.4800 NaN   \n",
       "4            Amplicon  ...    Male     29.0  NaN   139.680  35.4800 NaN   \n",
       "...               ...  ...     ...      ...  ...       ...      ...  ..   \n",
       "66575    Metagenomics  ...     NaN      0.7  NaN  -122.268  47.6614 NaN   \n",
       "66576    Metagenomics  ...     NaN      0.7  NaN  -122.268  47.6614 NaN   \n",
       "66577    Metagenomics  ...     NaN      0.7  NaN  -122.268  47.6614 NaN   \n",
       "66578    Metagenomics  ...     NaN      0.7  NaN  -122.268  47.6614 NaN   \n",
       "66579    Metagenomics  ...     NaN      0.7  NaN  -122.268  47.6614 NaN   \n",
       "\n",
       "      Recent.Antibiotics.Use antibiotics_used Antibiotics.Dose  \\\n",
       "0                        NaN              NaN              NaN   \n",
       "1                        NaN              NaN              NaN   \n",
       "2                        NaN              NaN              NaN   \n",
       "3                        NaN              NaN              NaN   \n",
       "4                        NaN              NaN              NaN   \n",
       "...                      ...              ...              ...   \n",
       "66575                    NaN              NaN              NaN   \n",
       "66576                    NaN              NaN              NaN   \n",
       "66577                    NaN              NaN              NaN   \n",
       "66578                    NaN              NaN              NaN   \n",
       "66579                    NaN              NaN              NaN   \n",
       "\n",
       "      Days.Without.Antibiotics.Use  \n",
       "0                              NaN  \n",
       "1                              NaN  \n",
       "2                              NaN  \n",
       "3                              NaN  \n",
       "4                              NaN  \n",
       "...                            ...  \n",
       "66575                          NaN  \n",
       "66576                          NaN  \n",
       "66577                          NaN  \n",
       "66578                          NaN  \n",
       "66579                          NaN  \n",
       "\n",
       "[66580 rows x 30 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "uid_map = pd.read_csv('samples_loaded.txt', delimiter = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uid</th>\n",
       "      <th>accession_id</th>\n",
       "      <th>data_type</th>\n",
       "      <th>tool_used</th>\n",
       "      <th>results_version</th>\n",
       "      <th>last_updated</th>\n",
       "      <th>QCStatus</th>\n",
       "      <th>QCMessage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>ERR525949</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 15:56:38</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>SRR413772</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 15:58:51</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>ERR526058</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 16:01:07</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>ERR209092</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-01-01 00:00:00</td>\n",
       "      <td>0</td>\n",
       "      <td>a single taxon unknown account for 100 percent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>ERR209066</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-01-01 00:00:00</td>\n",
       "      <td>0</td>\n",
       "      <td>a single taxon unknown account for 100 percent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52628</th>\n",
       "      <td>52855</td>\n",
       "      <td>SRR5548979</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52629</th>\n",
       "      <td>52856</td>\n",
       "      <td>SRR5548988</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52630</th>\n",
       "      <td>52857</td>\n",
       "      <td>SRR5548952</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52631</th>\n",
       "      <td>52858</td>\n",
       "      <td>SRR5548909</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52632</th>\n",
       "      <td>52859</td>\n",
       "      <td>SRR5548888</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52633 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         uid accession_id     data_type         tool_used  results_version  \\\n",
       "0          1    ERR525949  metagenomics  metaphlan ver2.0              0.1   \n",
       "1          2    SRR413772  metagenomics  metaphlan ver2.0              0.1   \n",
       "2          3    ERR526058  metagenomics  metaphlan ver2.0              0.1   \n",
       "3          4    ERR209092  metagenomics  metaphlan ver2.0              0.1   \n",
       "4          5    ERR209066  metagenomics  metaphlan ver2.0              0.1   \n",
       "...      ...          ...           ...               ...              ...   \n",
       "52628  52855   SRR5548979  metagenomics  metaphlan ver2.0              0.1   \n",
       "52629  52856   SRR5548988  metagenomics  metaphlan ver2.0              0.1   \n",
       "52630  52857   SRR5548952  metagenomics  metaphlan ver2.0              0.1   \n",
       "52631  52858   SRR5548909  metagenomics  metaphlan ver2.0              0.1   \n",
       "52632  52859   SRR5548888  metagenomics  metaphlan ver2.0              0.1   \n",
       "\n",
       "              last_updated  QCStatus  \\\n",
       "0      2018-06-22 15:56:38         1   \n",
       "1      2018-06-22 15:58:51         1   \n",
       "2      2018-06-22 16:01:07         1   \n",
       "3      2019-01-01 00:00:00         0   \n",
       "4      2019-01-01 00:00:00         0   \n",
       "...                    ...       ...   \n",
       "52628  2019-06-21 16:04:46         1   \n",
       "52629  2019-06-21 16:04:46         1   \n",
       "52630  2019-06-21 16:04:46         1   \n",
       "52631  2019-06-21 16:04:46         1   \n",
       "52632  2019-06-21 16:04:46         1   \n",
       "\n",
       "                                               QCMessage  \n",
       "0                                                    NaN  \n",
       "1                                                    NaN  \n",
       "2                                                    NaN  \n",
       "3      a single taxon unknown account for 100 percent...  \n",
       "4      a single taxon unknown account for 100 percent...  \n",
       "...                                                  ...  \n",
       "52628                                                NaN  \n",
       "52629                                                NaN  \n",
       "52630                                                NaN  \n",
       "52631                                                NaN  \n",
       "52632                                                NaN  \n",
       "\n",
       "[52633 rows x 8 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uid_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_metadata = pd.merge(uid_map, metadata, left_on='accession_id', right_on='run_id', how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uid</th>\n",
       "      <th>accession_id</th>\n",
       "      <th>data_type</th>\n",
       "      <th>tool_used</th>\n",
       "      <th>results_version</th>\n",
       "      <th>last_updated</th>\n",
       "      <th>QCStatus</th>\n",
       "      <th>QCMessage</th>\n",
       "      <th>checking</th>\n",
       "      <th>project_id</th>\n",
       "      <th>...</th>\n",
       "      <th>sex</th>\n",
       "      <th>host_age</th>\n",
       "      <th>diet</th>\n",
       "      <th>longitude</th>\n",
       "      <th>latitude</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Recent.Antibiotics.Use</th>\n",
       "      <th>antibiotics_used</th>\n",
       "      <th>Antibiotics.Dose</th>\n",
       "      <th>Days.Without.Antibiotics.Use</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>ERR525949</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 15:56:38</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJEB6456</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>SRR413772</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 15:58:51</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA422434</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>45.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2 months</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>ERR526058</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 16:01:07</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJEB6456</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>ERR209254</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 16:01:07</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJEB1220</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.5683</td>\n",
       "      <td>55.6761</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>ERR525843</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 16:01:07</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJEB6456</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53303</th>\n",
       "      <td>52855</td>\n",
       "      <td>SRR5548979</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA50637</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53304</th>\n",
       "      <td>52856</td>\n",
       "      <td>SRR5548988</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA50637</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53305</th>\n",
       "      <td>52857</td>\n",
       "      <td>SRR5548952</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA50637</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53306</th>\n",
       "      <td>52858</td>\n",
       "      <td>SRR5548909</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA50637</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53307</th>\n",
       "      <td>52859</td>\n",
       "      <td>SRR5548888</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA50637</td>\n",
       "      <td>...</td>\n",
       "      <td>Male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>53308 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         uid accession_id     data_type         tool_used  results_version  \\\n",
       "0          1    ERR525949  metagenomics  metaphlan ver2.0              0.1   \n",
       "1          2    SRR413772  metagenomics  metaphlan ver2.0              0.1   \n",
       "2          3    ERR526058  metagenomics  metaphlan ver2.0              0.1   \n",
       "3          6    ERR209254  metagenomics  metaphlan ver2.0              0.1   \n",
       "4          7    ERR525843  metagenomics  metaphlan ver2.0              0.1   \n",
       "...      ...          ...           ...               ...              ...   \n",
       "53303  52855   SRR5548979  metagenomics  metaphlan ver2.0              0.1   \n",
       "53304  52856   SRR5548988  metagenomics  metaphlan ver2.0              0.1   \n",
       "53305  52857   SRR5548952  metagenomics  metaphlan ver2.0              0.1   \n",
       "53306  52858   SRR5548909  metagenomics  metaphlan ver2.0              0.1   \n",
       "53307  52859   SRR5548888  metagenomics  metaphlan ver2.0              0.1   \n",
       "\n",
       "              last_updated  QCStatus QCMessage checking   project_id  ...  \\\n",
       "0      2018-06-22 15:56:38         1       NaN        1    PRJEB6456  ...   \n",
       "1      2018-06-22 15:58:51         1       NaN        1  PRJNA422434  ...   \n",
       "2      2018-06-22 16:01:07         1       NaN        1    PRJEB6456  ...   \n",
       "3      2018-06-22 16:01:07         1       NaN        1    PRJEB1220  ...   \n",
       "4      2018-06-22 16:01:07         1       NaN        1    PRJEB6456  ...   \n",
       "...                    ...       ...       ...      ...          ...  ...   \n",
       "53303  2019-06-21 16:04:46         1       NaN        1   PRJNA50637  ...   \n",
       "53304  2019-06-21 16:04:46         1       NaN        1   PRJNA50637  ...   \n",
       "53305  2019-06-21 16:04:46         1       NaN        1   PRJNA50637  ...   \n",
       "53306  2019-06-21 16:04:46         1       NaN        1   PRJNA50637  ...   \n",
       "53307  2019-06-21 16:04:46         1       NaN        1   PRJNA50637  ...   \n",
       "\n",
       "          sex host_age diet longitude latitude BMI Recent.Antibiotics.Use  \\\n",
       "0         NaN      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "1      Female     45.0  NaN       NaN      NaN NaN                      N   \n",
       "2         NaN      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "3         NaN      NaN  NaN   12.5683  55.6761 NaN                    NaN   \n",
       "4         NaN      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "...       ...      ...  ...       ...      ...  ..                    ...   \n",
       "53303  Female      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "53304  Female      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "53305  Female      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "53306  Female      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "53307    Male      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "\n",
       "      antibiotics_used  Antibiotics.Dose Days.Without.Antibiotics.Use  \n",
       "0                  NaN               NaN                          NaN  \n",
       "1                  NaN               NaN                     2 months  \n",
       "2                  NaN               NaN                          NaN  \n",
       "3                  NaN               NaN                          NaN  \n",
       "4                  NaN               NaN                          NaN  \n",
       "...                ...               ...                          ...  \n",
       "53303              NaN               NaN                          NaN  \n",
       "53304              NaN               NaN                          NaN  \n",
       "53305              NaN               NaN                          NaN  \n",
       "53306              NaN               NaN                          NaN  \n",
       "53307              NaN               NaN                          NaN  \n",
       "\n",
       "[53308 rows x 38 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_metadata"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are some samples that has more than one phenotype. For now, I'm filter these sample out from metadata but later we can keep them and do following instead: \n",
    "\n",
    "You need to transform your target labels into a multi-label format. For example, instead of having a single class label per sample, you represent the labels as a binary vector where each position corresponds to a phenotype, and 1 indicates the presence of that phenotype."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Find duplicate `uid`s\n",
    "duplicate_uids = merged_metadata[merged_metadata.duplicated('uid', keep=False)]['uid'].unique()\n",
    "\n",
    "# Drop rows with those duplicate `uid`s\n",
    "merged_metadata_cleaned = merged_metadata[~merged_metadata['uid'].isin(duplicate_uids)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26478\n"
     ]
    }
   ],
   "source": [
    "# Step 1: Find common UIDs between the two DataFrames\n",
    "common_uids = set(relative_abundance_df['loaded_uid']).intersection(set(merged_metadata_cleaned['uid']))\n",
    "\n",
    "print(len(common_uids))\n",
    "\n",
    "# Step 2: Filter both DataFrames to keep only the rows with common UIDs\n",
    "filtered_relative_abundance = relative_abundance_df[relative_abundance_df['loaded_uid'].isin(common_uids)]\n",
    "filtered_metadata = merged_metadata_cleaned[merged_metadata_cleaned['uid'].isin(common_uids)]\n",
    "\n",
    "# Step 3: Sort both DataFrames by the same order of UIDs\n",
    "# First, sort the common_uids list to define the order\n",
    "common_uids_sorted = sorted(common_uids)\n",
    "\n",
    "# Reindex both DataFrames to ensure they follow the same order\n",
    "filtered_relative_abundance = filtered_relative_abundance.set_index('loaded_uid').loc[common_uids_sorted].reset_index()\n",
    "filtered_metadata = filtered_metadata.set_index('uid').loc[common_uids_sorted].reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>loaded_uid</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>...</th>\n",
       "      <th>2211641</th>\n",
       "      <th>2212691</th>\n",
       "      <th>2212731</th>\n",
       "      <th>2282523</th>\n",
       "      <th>2282740</th>\n",
       "      <th>2282741</th>\n",
       "      <th>2282742</th>\n",
       "      <th>2304691</th>\n",
       "      <th>2304692</th>\n",
       "      <th>2529408</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26473</th>\n",
       "      <td>52855</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26474</th>\n",
       "      <td>52856</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26475</th>\n",
       "      <td>52857</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26476</th>\n",
       "      <td>52858</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26477</th>\n",
       "      <td>52859</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26478 rows × 8600 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       loaded_uid    6    7    9   10   11   13   14   16   17  ...  2211641  \\\n",
       "0               1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "1               2  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "2               3  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "3               6  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "4               7  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "...           ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...      ...   \n",
       "26473       52855  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "26474       52856  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "26475       52857  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "26476       52858  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "26477       52859  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...      0.0   \n",
       "\n",
       "       2212691  2212731  2282523  2282740  2282741  2282742  2304691  2304692  \\\n",
       "0          0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "1          0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "2          0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "3          0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "4          0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "...        ...      ...      ...      ...      ...      ...      ...      ...   \n",
       "26473      0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "26474      0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "26475      0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "26476      0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "26477      0.0      0.0      0.0      0.0      0.0      0.0      0.0      0.0   \n",
       "\n",
       "       2529408  \n",
       "0          0.0  \n",
       "1          0.0  \n",
       "2          0.0  \n",
       "3          0.0  \n",
       "4          0.0  \n",
       "...        ...  \n",
       "26473      0.0  \n",
       "26474      0.0  \n",
       "26475      0.0  \n",
       "26476      0.0  \n",
       "26477      0.0  \n",
       "\n",
       "[26478 rows x 8600 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_relative_abundance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uid</th>\n",
       "      <th>accession_id</th>\n",
       "      <th>data_type</th>\n",
       "      <th>tool_used</th>\n",
       "      <th>results_version</th>\n",
       "      <th>last_updated</th>\n",
       "      <th>QCStatus</th>\n",
       "      <th>QCMessage</th>\n",
       "      <th>checking</th>\n",
       "      <th>project_id</th>\n",
       "      <th>...</th>\n",
       "      <th>sex</th>\n",
       "      <th>host_age</th>\n",
       "      <th>diet</th>\n",
       "      <th>longitude</th>\n",
       "      <th>latitude</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Recent.Antibiotics.Use</th>\n",
       "      <th>antibiotics_used</th>\n",
       "      <th>Antibiotics.Dose</th>\n",
       "      <th>Days.Without.Antibiotics.Use</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>ERR525949</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 15:56:38</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJEB6456</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>SRR413772</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 15:58:51</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA422434</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>45.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2 months</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>ERR526058</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 16:01:07</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJEB6456</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>ERR209254</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 16:01:07</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJEB1220</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.5683</td>\n",
       "      <td>55.6761</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>ERR525843</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 16:01:07</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJEB6456</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26473</th>\n",
       "      <td>52855</td>\n",
       "      <td>SRR5548979</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA50637</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26474</th>\n",
       "      <td>52856</td>\n",
       "      <td>SRR5548988</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA50637</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26475</th>\n",
       "      <td>52857</td>\n",
       "      <td>SRR5548952</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA50637</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26476</th>\n",
       "      <td>52858</td>\n",
       "      <td>SRR5548909</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA50637</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26477</th>\n",
       "      <td>52859</td>\n",
       "      <td>SRR5548888</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA50637</td>\n",
       "      <td>...</td>\n",
       "      <td>Male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26478 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         uid accession_id     data_type         tool_used  results_version  \\\n",
       "0          1    ERR525949  metagenomics  metaphlan ver2.0              0.1   \n",
       "1          2    SRR413772  metagenomics  metaphlan ver2.0              0.1   \n",
       "2          3    ERR526058  metagenomics  metaphlan ver2.0              0.1   \n",
       "3          6    ERR209254  metagenomics  metaphlan ver2.0              0.1   \n",
       "4          7    ERR525843  metagenomics  metaphlan ver2.0              0.1   \n",
       "...      ...          ...           ...               ...              ...   \n",
       "26473  52855   SRR5548979  metagenomics  metaphlan ver2.0              0.1   \n",
       "26474  52856   SRR5548988  metagenomics  metaphlan ver2.0              0.1   \n",
       "26475  52857   SRR5548952  metagenomics  metaphlan ver2.0              0.1   \n",
       "26476  52858   SRR5548909  metagenomics  metaphlan ver2.0              0.1   \n",
       "26477  52859   SRR5548888  metagenomics  metaphlan ver2.0              0.1   \n",
       "\n",
       "              last_updated  QCStatus QCMessage checking   project_id  ...  \\\n",
       "0      2018-06-22 15:56:38         1       NaN        1    PRJEB6456  ...   \n",
       "1      2018-06-22 15:58:51         1       NaN        1  PRJNA422434  ...   \n",
       "2      2018-06-22 16:01:07         1       NaN        1    PRJEB6456  ...   \n",
       "3      2018-06-22 16:01:07         1       NaN        1    PRJEB1220  ...   \n",
       "4      2018-06-22 16:01:07         1       NaN        1    PRJEB6456  ...   \n",
       "...                    ...       ...       ...      ...          ...  ...   \n",
       "26473  2019-06-21 16:04:46         1       NaN        1   PRJNA50637  ...   \n",
       "26474  2019-06-21 16:04:46         1       NaN        1   PRJNA50637  ...   \n",
       "26475  2019-06-21 16:04:46         1       NaN        1   PRJNA50637  ...   \n",
       "26476  2019-06-21 16:04:46         1       NaN        1   PRJNA50637  ...   \n",
       "26477  2019-06-21 16:04:46         1       NaN        1   PRJNA50637  ...   \n",
       "\n",
       "          sex host_age diet longitude latitude BMI Recent.Antibiotics.Use  \\\n",
       "0         NaN      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "1      Female     45.0  NaN       NaN      NaN NaN                      N   \n",
       "2         NaN      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "3         NaN      NaN  NaN   12.5683  55.6761 NaN                    NaN   \n",
       "4         NaN      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "...       ...      ...  ...       ...      ...  ..                    ...   \n",
       "26473  Female      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "26474  Female      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "26475  Female      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "26476  Female      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "26477    Male      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "\n",
       "      antibiotics_used  Antibiotics.Dose Days.Without.Antibiotics.Use  \n",
       "0                  NaN               NaN                          NaN  \n",
       "1                  NaN               NaN                     2 months  \n",
       "2                  NaN               NaN                          NaN  \n",
       "3                  NaN               NaN                          NaN  \n",
       "4                  NaN               NaN                          NaN  \n",
       "...                ...               ...                          ...  \n",
       "26473              NaN               NaN                          NaN  \n",
       "26474              NaN               NaN                          NaN  \n",
       "26475              NaN               NaN                          NaN  \n",
       "26476              NaN               NaN                          NaN  \n",
       "26477              NaN               NaN                          NaN  \n",
       "\n",
       "[26478 rows x 38 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "filtered_relative_abundance.to_csv('filtered_relative_abundace.csv', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "filtered_metadata.to_csv('filtered_metadata.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### EDA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3045288/4032808118.py:2: DtypeWarning: Columns (10) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  metadata = pd.read_csv('filtered_metadata.csv')\n"
     ]
    }
   ],
   "source": [
    "# relative_abundance = pd.read_csv('filtered_relative_abundance.csv')\n",
    "metadata = pd.read_csv('filtered_metadata.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['uid', 'accession_id', 'data_type', 'tool_used', 'results_version',\n",
       "       'last_updated', 'QCStatus', 'QCMessage', 'checking', 'project_id',\n",
       "       'our_project_id', 'sample_name', 'original_sample_description',\n",
       "       'curated_sample_description', 'run_id', 'sample_id', 'second_sample_id',\n",
       "       'experiment_type', 'nr_reads_sequenced', 'instrument_model', 'disease',\n",
       "       'phenotype', 'is_disease_stage_available', 'disease_stage', 'more',\n",
       "       'more_info', 'country', 'collection_date', 'sex', 'host_age', 'diet',\n",
       "       'longitude', 'latitude', 'BMI', 'Recent.Antibiotics.Use',\n",
       "       'antibiotics_used', 'Antibiotics.Dose', 'Days.Without.Antibiotics.Use'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uid</th>\n",
       "      <th>accession_id</th>\n",
       "      <th>data_type</th>\n",
       "      <th>tool_used</th>\n",
       "      <th>results_version</th>\n",
       "      <th>last_updated</th>\n",
       "      <th>QCStatus</th>\n",
       "      <th>QCMessage</th>\n",
       "      <th>checking</th>\n",
       "      <th>project_id</th>\n",
       "      <th>...</th>\n",
       "      <th>sex</th>\n",
       "      <th>host_age</th>\n",
       "      <th>diet</th>\n",
       "      <th>longitude</th>\n",
       "      <th>latitude</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Recent.Antibiotics.Use</th>\n",
       "      <th>antibiotics_used</th>\n",
       "      <th>Antibiotics.Dose</th>\n",
       "      <th>Days.Without.Antibiotics.Use</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>ERR525949</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 15:56:38</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJEB6456</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>SRR413772</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 15:58:51</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA422434</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>45.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2 months</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>ERR526058</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 16:01:07</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJEB6456</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>ERR209254</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 16:01:07</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJEB1220</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.5683</td>\n",
       "      <td>55.6761</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>ERR525843</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2018-06-22 16:01:07</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJEB6456</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26473</th>\n",
       "      <td>52855</td>\n",
       "      <td>SRR5548979</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA50637</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26474</th>\n",
       "      <td>52856</td>\n",
       "      <td>SRR5548988</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA50637</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26475</th>\n",
       "      <td>52857</td>\n",
       "      <td>SRR5548952</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA50637</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26476</th>\n",
       "      <td>52858</td>\n",
       "      <td>SRR5548909</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA50637</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26477</th>\n",
       "      <td>52859</td>\n",
       "      <td>SRR5548888</td>\n",
       "      <td>metagenomics</td>\n",
       "      <td>metaphlan ver2.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2019-06-21 16:04:46</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA50637</td>\n",
       "      <td>...</td>\n",
       "      <td>Male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26478 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         uid accession_id     data_type         tool_used  results_version  \\\n",
       "0          1    ERR525949  metagenomics  metaphlan ver2.0              0.1   \n",
       "1          2    SRR413772  metagenomics  metaphlan ver2.0              0.1   \n",
       "2          3    ERR526058  metagenomics  metaphlan ver2.0              0.1   \n",
       "3          6    ERR209254  metagenomics  metaphlan ver2.0              0.1   \n",
       "4          7    ERR525843  metagenomics  metaphlan ver2.0              0.1   \n",
       "...      ...          ...           ...               ...              ...   \n",
       "26473  52855   SRR5548979  metagenomics  metaphlan ver2.0              0.1   \n",
       "26474  52856   SRR5548988  metagenomics  metaphlan ver2.0              0.1   \n",
       "26475  52857   SRR5548952  metagenomics  metaphlan ver2.0              0.1   \n",
       "26476  52858   SRR5548909  metagenomics  metaphlan ver2.0              0.1   \n",
       "26477  52859   SRR5548888  metagenomics  metaphlan ver2.0              0.1   \n",
       "\n",
       "              last_updated  QCStatus  QCMessage  checking   project_id  ...  \\\n",
       "0      2018-06-22 15:56:38         1        NaN         1    PRJEB6456  ...   \n",
       "1      2018-06-22 15:58:51         1        NaN         1  PRJNA422434  ...   \n",
       "2      2018-06-22 16:01:07         1        NaN         1    PRJEB6456  ...   \n",
       "3      2018-06-22 16:01:07         1        NaN         1    PRJEB1220  ...   \n",
       "4      2018-06-22 16:01:07         1        NaN         1    PRJEB6456  ...   \n",
       "...                    ...       ...        ...       ...          ...  ...   \n",
       "26473  2019-06-21 16:04:46         1        NaN         1   PRJNA50637  ...   \n",
       "26474  2019-06-21 16:04:46         1        NaN         1   PRJNA50637  ...   \n",
       "26475  2019-06-21 16:04:46         1        NaN         1   PRJNA50637  ...   \n",
       "26476  2019-06-21 16:04:46         1        NaN         1   PRJNA50637  ...   \n",
       "26477  2019-06-21 16:04:46         1        NaN         1   PRJNA50637  ...   \n",
       "\n",
       "          sex host_age diet longitude latitude BMI Recent.Antibiotics.Use  \\\n",
       "0         NaN      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "1      Female     45.0  NaN       NaN      NaN NaN                      N   \n",
       "2         NaN      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "3         NaN      NaN  NaN   12.5683  55.6761 NaN                    NaN   \n",
       "4         NaN      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "...       ...      ...  ...       ...      ...  ..                    ...   \n",
       "26473  Female      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "26474  Female      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "26475  Female      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "26476  Female      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "26477    Male      NaN  NaN       NaN      NaN NaN                    NaN   \n",
       "\n",
       "      antibiotics_used  Antibiotics.Dose Days.Without.Antibiotics.Use  \n",
       "0                  NaN               NaN                          NaN  \n",
       "1                  NaN               NaN                     2 months  \n",
       "2                  NaN               NaN                          NaN  \n",
       "3                  NaN               NaN                          NaN  \n",
       "4                  NaN               NaN                          NaN  \n",
       "...                ...               ...                          ...  \n",
       "26473              NaN               NaN                          NaN  \n",
       "26474              NaN               NaN                          NaN  \n",
       "26475              NaN               NaN                          NaN  \n",
       "26476              NaN               NaN                          NaN  \n",
       "26477              NaN               NaN                          NaN  \n",
       "\n",
       "[26478 rows x 38 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "sex\n",
       "Male      6122\n",
       "Female    5756\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata['sex'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata['disease'].value_counts().to_csv('disease_count.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata['project_id'].value_counts().to_csv('projects_count.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### IBD data filtering (train and test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3045702/3371440668.py:1: DtypeWarning: Columns (0,2,5,14,15,22,26,27,28,29) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  metadata = pd.read_csv('sample_to_run_info.txt', delimiter = '\\t')\n"
     ]
    }
   ],
   "source": [
    "metadata = pd.read_csv('sample_to_run_info.txt', delimiter = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>checking</th>\n",
       "      <th>project_id</th>\n",
       "      <th>our_project_id</th>\n",
       "      <th>sample_name</th>\n",
       "      <th>original_sample_description</th>\n",
       "      <th>curated_sample_description</th>\n",
       "      <th>run_id</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>second_sample_id</th>\n",
       "      <th>experiment_type</th>\n",
       "      <th>...</th>\n",
       "      <th>sex</th>\n",
       "      <th>host_age</th>\n",
       "      <th>diet</th>\n",
       "      <th>longitude</th>\n",
       "      <th>latitude</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Recent.Antibiotics.Use</th>\n",
       "      <th>antibiotics_used</th>\n",
       "      <th>Antibiotics.Dose</th>\n",
       "      <th>Days.Without.Antibiotics.Use</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJDB4360</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Japanese001</td>\n",
       "      <td>Healthy Japanese gut microbiota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DRR048993</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD00042843</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>28.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>139.680</td>\n",
       "      <td>35.4800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJDB4360</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Japanese002</td>\n",
       "      <td>Healthy Japanese gut microbiota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DRR048994</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD00042844</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>Male</td>\n",
       "      <td>31.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>139.680</td>\n",
       "      <td>35.4800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJDB4360</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Japanese003</td>\n",
       "      <td>Healthy Japanese gut microbiota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DRR048995</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD00042845</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>Male</td>\n",
       "      <td>60.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>139.680</td>\n",
       "      <td>35.4800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJDB4360</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Japanese004</td>\n",
       "      <td>Healthy Japanese gut microbiota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DRR048996</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD00042846</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>Male</td>\n",
       "      <td>57.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>139.680</td>\n",
       "      <td>35.4800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJDB4360</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Japanese005</td>\n",
       "      <td>Healthy Japanese gut microbiota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DRR048997</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD00042847</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>Male</td>\n",
       "      <td>29.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>139.680</td>\n",
       "      <td>35.4800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66575</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA215106</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F34_CY</td>\n",
       "      <td>Functional metagenomic selections were perform...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR961853</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>Metagenomics</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-122.268</td>\n",
       "      <td>47.6614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66576</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA215106</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F34_PE</td>\n",
       "      <td>Functional metagenomic selections were perform...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR961854</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>Metagenomics</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-122.268</td>\n",
       "      <td>47.6614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66577</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA215106</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F34_TE</td>\n",
       "      <td>Functional metagenomic selections were perform...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR961855</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>Metagenomics</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-122.268</td>\n",
       "      <td>47.6614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66578</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA215106</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F34_TR</td>\n",
       "      <td>Functional metagenomic selections were perform...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR961856</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>Metagenomics</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-122.268</td>\n",
       "      <td>47.6614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66579</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA215106</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F34_TRSX</td>\n",
       "      <td>Functional metagenomic selections were perform...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR961857</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>SAMN02317178</td>\n",
       "      <td>Metagenomics</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-122.268</td>\n",
       "      <td>47.6614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>66580 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      checking   project_id our_project_id  sample_name  \\\n",
       "0            1    PRJDB4360            NaN  Japanese001   \n",
       "1            1    PRJDB4360            NaN  Japanese002   \n",
       "2            1    PRJDB4360            NaN  Japanese003   \n",
       "3            1    PRJDB4360            NaN  Japanese004   \n",
       "4            1    PRJDB4360            NaN  Japanese005   \n",
       "...        ...          ...            ...          ...   \n",
       "66575        1  PRJNA215106            NaN       F34_CY   \n",
       "66576        1  PRJNA215106            NaN       F34_PE   \n",
       "66577        1  PRJNA215106            NaN       F34_TE   \n",
       "66578        1  PRJNA215106            NaN       F34_TR   \n",
       "66579        1  PRJNA215106            NaN     F34_TRSX   \n",
       "\n",
       "                             original_sample_description  \\\n",
       "0                        Healthy Japanese gut microbiota   \n",
       "1                        Healthy Japanese gut microbiota   \n",
       "2                        Healthy Japanese gut microbiota   \n",
       "3                        Healthy Japanese gut microbiota   \n",
       "4                        Healthy Japanese gut microbiota   \n",
       "...                                                  ...   \n",
       "66575  Functional metagenomic selections were perform...   \n",
       "66576  Functional metagenomic selections were perform...   \n",
       "66577  Functional metagenomic selections were perform...   \n",
       "66578  Functional metagenomic selections were perform...   \n",
       "66579  Functional metagenomic selections were perform...   \n",
       "\n",
       "      curated_sample_description     run_id     sample_id second_sample_id  \\\n",
       "0                            NaN  DRR048993           NaN     SAMD00042843   \n",
       "1                            NaN  DRR048994           NaN     SAMD00042844   \n",
       "2                            NaN  DRR048995           NaN     SAMD00042845   \n",
       "3                            NaN  DRR048996           NaN     SAMD00042846   \n",
       "4                            NaN  DRR048997           NaN     SAMD00042847   \n",
       "...                          ...        ...           ...              ...   \n",
       "66575                        NaN  SRR961853  SAMN02317178     SAMN02317178   \n",
       "66576                        NaN  SRR961854  SAMN02317178     SAMN02317178   \n",
       "66577                        NaN  SRR961855  SAMN02317178     SAMN02317178   \n",
       "66578                        NaN  SRR961856  SAMN02317178     SAMN02317178   \n",
       "66579                        NaN  SRR961857  SAMN02317178     SAMN02317178   \n",
       "\n",
       "      experiment_type  ...     sex host_age diet longitude latitude BMI  \\\n",
       "0            Amplicon  ...  Female     28.0  NaN   139.680  35.4800 NaN   \n",
       "1            Amplicon  ...    Male     31.0  NaN   139.680  35.4800 NaN   \n",
       "2            Amplicon  ...    Male     60.0  NaN   139.680  35.4800 NaN   \n",
       "3            Amplicon  ...    Male     57.0  NaN   139.680  35.4800 NaN   \n",
       "4            Amplicon  ...    Male     29.0  NaN   139.680  35.4800 NaN   \n",
       "...               ...  ...     ...      ...  ...       ...      ...  ..   \n",
       "66575    Metagenomics  ...     NaN      0.7  NaN  -122.268  47.6614 NaN   \n",
       "66576    Metagenomics  ...     NaN      0.7  NaN  -122.268  47.6614 NaN   \n",
       "66577    Metagenomics  ...     NaN      0.7  NaN  -122.268  47.6614 NaN   \n",
       "66578    Metagenomics  ...     NaN      0.7  NaN  -122.268  47.6614 NaN   \n",
       "66579    Metagenomics  ...     NaN      0.7  NaN  -122.268  47.6614 NaN   \n",
       "\n",
       "      Recent.Antibiotics.Use antibiotics_used Antibiotics.Dose  \\\n",
       "0                        NaN              NaN              NaN   \n",
       "1                        NaN              NaN              NaN   \n",
       "2                        NaN              NaN              NaN   \n",
       "3                        NaN              NaN              NaN   \n",
       "4                        NaN              NaN              NaN   \n",
       "...                      ...              ...              ...   \n",
       "66575                    NaN              NaN              NaN   \n",
       "66576                    NaN              NaN              NaN   \n",
       "66577                    NaN              NaN              NaN   \n",
       "66578                    NaN              NaN              NaN   \n",
       "66579                    NaN              NaN              NaN   \n",
       "\n",
       "      Days.Without.Antibiotics.Use  \n",
       "0                              NaN  \n",
       "1                              NaN  \n",
       "2                              NaN  \n",
       "3                              NaN  \n",
       "4                              NaN  \n",
       "...                            ...  \n",
       "66575                          NaN  \n",
       "66576                          NaN  \n",
       "66577                          NaN  \n",
       "66578                          NaN  \n",
       "66579                          NaN  \n",
       "\n",
       "[66580 rows x 30 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I found IBD projects on GMrepo following link: https://gmrepo.humangut.info/phenotypes/D043183"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "IBD_projects = ['PRJEB11419', 'PRJNA268708', 'PRJNA46339', 'PRJNA302437', 'PRJNA392762', 'PRJNA386442', 'PRJDB5442', 'PRJNA373876', 'PRJNA524547' ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "Filtered_projects_metadata = metadata[metadata['project_id'].isin(IBD_projects)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "diseases = ['D043183', 'D006262']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "Filtered_projects_disease_metadata = Filtered_projects_metadata[Filtered_projects_metadata['disease'].isin(diseases)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>checking</th>\n",
       "      <th>project_id</th>\n",
       "      <th>our_project_id</th>\n",
       "      <th>sample_name</th>\n",
       "      <th>original_sample_description</th>\n",
       "      <th>curated_sample_description</th>\n",
       "      <th>run_id</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>second_sample_id</th>\n",
       "      <th>experiment_type</th>\n",
       "      <th>...</th>\n",
       "      <th>sex</th>\n",
       "      <th>host_age</th>\n",
       "      <th>diet</th>\n",
       "      <th>longitude</th>\n",
       "      <th>latitude</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Recent.Antibiotics.Use</th>\n",
       "      <th>antibiotics_used</th>\n",
       "      <th>Antibiotics.Dose</th>\n",
       "      <th>Days.Without.Antibiotics.Use</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1192</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJDB5442</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FGIDs01-D01.3000.fastq</td>\n",
       "      <td>Stool sample of donor used for fecal microbiot...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DRR086903</td>\n",
       "      <td>DRS058942</td>\n",
       "      <td>SAMD00074226</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1193</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJDB5442</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FGIDs01-R00w.3000.fastq</td>\n",
       "      <td>Stool sample of patients at week 0 after fecal...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DRR086904</td>\n",
       "      <td>DRS058943</td>\n",
       "      <td>SAMD00074227</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1194</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJDB5442</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FGIDs01-R04w.3000.fastq</td>\n",
       "      <td>Stool sample of patients at week 4 after fecal...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DRR086905</td>\n",
       "      <td>DRS058944</td>\n",
       "      <td>SAMD00074228</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJDB5442</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FGIDs01-R12w.3000.fastq</td>\n",
       "      <td>Stool sample of patients at week 12 after feca...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DRR086906</td>\n",
       "      <td>DRS058945</td>\n",
       "      <td>SAMD00074229</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1196</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJDB5442</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FGIDs03-D01.3000.fastq</td>\n",
       "      <td>Stool sample of donor used for fecal microbiot...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DRR086907</td>\n",
       "      <td>DRS058946</td>\n",
       "      <td>SAMD00074230</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56120</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA392762</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FIBS_BS41</td>\n",
       "      <td>MIMS Environmental/Metagenome sample from huma...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR5796813</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMN07309937</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>39.0</td>\n",
       "      <td>Control</td>\n",
       "      <td>174.89</td>\n",
       "      <td>-40.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56121</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA392762</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FIBS_BS44</td>\n",
       "      <td>MIMS Environmental/Metagenome sample from huma...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR5796814</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMN07309940</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>Male</td>\n",
       "      <td>46.0</td>\n",
       "      <td>LowFODMAP</td>\n",
       "      <td>174.89</td>\n",
       "      <td>-40.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56122</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA392762</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FIBS_BS43</td>\n",
       "      <td>MIMS Environmental/Metagenome sample from huma...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR5796815</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMN07309939</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>32.0</td>\n",
       "      <td>Control</td>\n",
       "      <td>174.89</td>\n",
       "      <td>-40.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56123</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA392762</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FIBS_BS50</td>\n",
       "      <td>MIMS Environmental/Metagenome sample from huma...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR5796816</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMN07309946</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>49.0</td>\n",
       "      <td>LowFODMAP</td>\n",
       "      <td>174.89</td>\n",
       "      <td>-40.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56124</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA392762</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FIBS_BS49</td>\n",
       "      <td>MIMS Environmental/Metagenome sample from huma...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR5796817</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMN07309945</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>174.89</td>\n",
       "      <td>-40.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5252 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      checking   project_id our_project_id              sample_name  \\\n",
       "1192         1    PRJDB5442            NaN   FGIDs01-D01.3000.fastq   \n",
       "1193         1    PRJDB5442            NaN  FGIDs01-R00w.3000.fastq   \n",
       "1194         1    PRJDB5442            NaN  FGIDs01-R04w.3000.fastq   \n",
       "1195         1    PRJDB5442            NaN  FGIDs01-R12w.3000.fastq   \n",
       "1196         1    PRJDB5442            NaN   FGIDs03-D01.3000.fastq   \n",
       "...        ...          ...            ...                      ...   \n",
       "56120        1  PRJNA392762            NaN                FIBS_BS41   \n",
       "56121        1  PRJNA392762            NaN                FIBS_BS44   \n",
       "56122        1  PRJNA392762            NaN                FIBS_BS43   \n",
       "56123        1  PRJNA392762            NaN                FIBS_BS50   \n",
       "56124        1  PRJNA392762            NaN                FIBS_BS49   \n",
       "\n",
       "                             original_sample_description  \\\n",
       "1192   Stool sample of donor used for fecal microbiot...   \n",
       "1193   Stool sample of patients at week 0 after fecal...   \n",
       "1194   Stool sample of patients at week 4 after fecal...   \n",
       "1195   Stool sample of patients at week 12 after feca...   \n",
       "1196   Stool sample of donor used for fecal microbiot...   \n",
       "...                                                  ...   \n",
       "56120  MIMS Environmental/Metagenome sample from huma...   \n",
       "56121  MIMS Environmental/Metagenome sample from huma...   \n",
       "56122  MIMS Environmental/Metagenome sample from huma...   \n",
       "56123  MIMS Environmental/Metagenome sample from huma...   \n",
       "56124  MIMS Environmental/Metagenome sample from huma...   \n",
       "\n",
       "      curated_sample_description      run_id  sample_id second_sample_id  \\\n",
       "1192                         NaN   DRR086903  DRS058942     SAMD00074226   \n",
       "1193                         NaN   DRR086904  DRS058943     SAMD00074227   \n",
       "1194                         NaN   DRR086905  DRS058944     SAMD00074228   \n",
       "1195                         NaN   DRR086906  DRS058945     SAMD00074229   \n",
       "1196                         NaN   DRR086907  DRS058946     SAMD00074230   \n",
       "...                          ...         ...        ...              ...   \n",
       "56120                        NaN  SRR5796813        NaN     SAMN07309937   \n",
       "56121                        NaN  SRR5796814        NaN     SAMN07309940   \n",
       "56122                        NaN  SRR5796815        NaN     SAMN07309939   \n",
       "56123                        NaN  SRR5796816        NaN     SAMN07309946   \n",
       "56124                        NaN  SRR5796817        NaN     SAMN07309945   \n",
       "\n",
       "      experiment_type  ...     sex host_age       diet longitude latitude BMI  \\\n",
       "1192         Amplicon  ...     NaN      NaN        NaN       NaN      NaN NaN   \n",
       "1193         Amplicon  ...     NaN      NaN        NaN       NaN      NaN NaN   \n",
       "1194         Amplicon  ...     NaN      NaN        NaN       NaN      NaN NaN   \n",
       "1195         Amplicon  ...     NaN      NaN        NaN       NaN      NaN NaN   \n",
       "1196         Amplicon  ...     NaN      NaN        NaN       NaN      NaN NaN   \n",
       "...               ...  ...     ...      ...        ...       ...      ...  ..   \n",
       "56120        Amplicon  ...  Female     39.0    Control    174.89    -40.9 NaN   \n",
       "56121        Amplicon  ...    Male     46.0  LowFODMAP    174.89    -40.9 NaN   \n",
       "56122        Amplicon  ...  Female     32.0    Control    174.89    -40.9 NaN   \n",
       "56123        Amplicon  ...  Female     49.0  LowFODMAP    174.89    -40.9 NaN   \n",
       "56124        Amplicon  ...     NaN      NaN        NaN    174.89    -40.9 NaN   \n",
       "\n",
       "      Recent.Antibiotics.Use antibiotics_used Antibiotics.Dose  \\\n",
       "1192                     NaN              NaN              NaN   \n",
       "1193                     NaN              NaN              NaN   \n",
       "1194                     NaN              NaN              NaN   \n",
       "1195                     NaN              NaN              NaN   \n",
       "1196                     NaN              NaN              NaN   \n",
       "...                      ...              ...              ...   \n",
       "56120                    NaN              NaN              NaN   \n",
       "56121                    NaN              NaN              NaN   \n",
       "56122                    NaN              NaN              NaN   \n",
       "56123                    NaN              NaN              NaN   \n",
       "56124                    NaN              NaN              NaN   \n",
       "\n",
       "      Days.Without.Antibiotics.Use  \n",
       "1192                           NaN  \n",
       "1193                           NaN  \n",
       "1194                           NaN  \n",
       "1195                           NaN  \n",
       "1196                           NaN  \n",
       "...                            ...  \n",
       "56120                          NaN  \n",
       "56121                          NaN  \n",
       "56122                          NaN  \n",
       "56123                          NaN  \n",
       "56124                          NaN  \n",
       "\n",
       "[5252 rows x 30 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Filtered_projects_disease_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "project_id\n",
       "PRJEB11419     4205\n",
       "PRJNA46339      355\n",
       "PRJNA268708     290\n",
       "PRJNA302437     152\n",
       "PRJNA392762     110\n",
       "PRJNA373876      52\n",
       "PRJNA386442      48\n",
       "PRJDB5442        40\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Filtered_projects_disease_metadata['project_id'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "uid_map = pd.read_csv('samples_loaded.txt', delimiter = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_metadata_IBD = pd.merge(uid_map, Filtered_projects_disease_metadata, left_on='accession_id', right_on='run_id', how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5251, 38)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_metadata_IBD.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "relative_abundance_df = pd.read_csv('relative_abundance.csv', index_col = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2441\n"
     ]
    }
   ],
   "source": [
    "# Step 1: Find common UIDs between the two DataFrames\n",
    "common_uids_IBD = set(relative_abundance_df['loaded_uid']).intersection(set(merged_metadata_IBD['uid']))\n",
    "\n",
    "print(len(common_uids_IBD))\n",
    "\n",
    "# Step 2: Filter both DataFrames to keep only the rows with common UIDs\n",
    "filtered_relative_abundance_IBD = relative_abundance_df[relative_abundance_df['loaded_uid'].isin(common_uids_IBD)]\n",
    "filtered_metadata_IBD = merged_metadata_IBD[merged_metadata_IBD['uid'].isin(common_uids_IBD)]\n",
    "\n",
    "# Step 3: Sort both DataFrames by the same order of UIDs\n",
    "# First, sort the common_uids list to define the order\n",
    "common_uids_IBD_sorted = sorted(common_uids_IBD)\n",
    "\n",
    "# Reindex both DataFrames to ensure they follow the same order\n",
    "filtered_relative_abundance_IBD = filtered_relative_abundance_IBD.set_index('loaded_uid').loc[common_uids_IBD_sorted].reset_index()\n",
    "filtered_metadata_IBD = filtered_metadata_IBD.set_index('uid').loc[common_uids_IBD_sorted].reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "filtered_metadata_IBD.to_csv('filtered_metadata_IBD.csv', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "filtered_relative_abundance_IBD.to_csv('filtered_relative_abundance_IBD.csv', index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### IBD metadata EDA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_IBD = pd.read_csv('filtered_metadata_IBD.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uid</th>\n",
       "      <th>accession_id</th>\n",
       "      <th>data_type</th>\n",
       "      <th>tool_used</th>\n",
       "      <th>results_version</th>\n",
       "      <th>last_updated</th>\n",
       "      <th>QCStatus</th>\n",
       "      <th>QCMessage</th>\n",
       "      <th>checking</th>\n",
       "      <th>project_id</th>\n",
       "      <th>...</th>\n",
       "      <th>sex</th>\n",
       "      <th>host_age</th>\n",
       "      <th>diet</th>\n",
       "      <th>longitude</th>\n",
       "      <th>latitude</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Recent.Antibiotics.Use</th>\n",
       "      <th>antibiotics_used</th>\n",
       "      <th>Antibiotics.Dose</th>\n",
       "      <th>Days.Without.Antibiotics.Use</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4721</td>\n",
       "      <td>SRR1686980</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>Mapseq</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2019-01-01 00:00:00</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA268708</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.1833</td>\n",
       "      <td>41.3833</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4722</td>\n",
       "      <td>SRR1686981</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>Mapseq</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2019-01-01 00:00:00</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA268708</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.1833</td>\n",
       "      <td>41.3833</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4723</td>\n",
       "      <td>SRR1686982</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>Mapseq</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2019-01-01 00:00:00</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA268708</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.1833</td>\n",
       "      <td>41.3833</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4724</td>\n",
       "      <td>SRR1687009</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>Mapseq</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2019-01-01 00:00:00</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA268708</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.1833</td>\n",
       "      <td>41.3833</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4725</td>\n",
       "      <td>SRR1687039</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>Mapseq</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2019-01-01 00:00:00</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA268708</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.1833</td>\n",
       "      <td>41.3833</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2436</th>\n",
       "      <td>50638</td>\n",
       "      <td>SRR051708</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>Mapseq</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2019-06-05 10:02:31</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA46339</td>\n",
       "      <td>...</td>\n",
       "      <td>Male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2437</th>\n",
       "      <td>50639</td>\n",
       "      <td>SRR051709</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>Mapseq</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2019-06-05 10:02:31</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA46339</td>\n",
       "      <td>...</td>\n",
       "      <td>Male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2438</th>\n",
       "      <td>50640</td>\n",
       "      <td>SRR051710</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>Mapseq</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2019-06-05 10:02:31</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA46339</td>\n",
       "      <td>...</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2439</th>\n",
       "      <td>50641</td>\n",
       "      <td>SRR051715</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>Mapseq</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2019-06-05 10:02:31</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA46339</td>\n",
       "      <td>...</td>\n",
       "      <td>Male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2440</th>\n",
       "      <td>50642</td>\n",
       "      <td>SRR051747</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>Mapseq</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2019-06-05 10:02:31</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PRJNA46339</td>\n",
       "      <td>...</td>\n",
       "      <td>Male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2441 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        uid accession_id data_type tool_used  results_version  \\\n",
       "0      4721   SRR1686980  amplicon    Mapseq              1.0   \n",
       "1      4722   SRR1686981  amplicon    Mapseq              1.0   \n",
       "2      4723   SRR1686982  amplicon    Mapseq              1.0   \n",
       "3      4724   SRR1687009  amplicon    Mapseq              1.0   \n",
       "4      4725   SRR1687039  amplicon    Mapseq              1.0   \n",
       "...     ...          ...       ...       ...              ...   \n",
       "2436  50638    SRR051708  amplicon    Mapseq              1.0   \n",
       "2437  50639    SRR051709  amplicon    Mapseq              1.0   \n",
       "2438  50640    SRR051710  amplicon    Mapseq              1.0   \n",
       "2439  50641    SRR051715  amplicon    Mapseq              1.0   \n",
       "2440  50642    SRR051747  amplicon    Mapseq              1.0   \n",
       "\n",
       "             last_updated  QCStatus  QCMessage  checking   project_id  ...  \\\n",
       "0     2019-01-01 00:00:00         1        NaN         1  PRJNA268708  ...   \n",
       "1     2019-01-01 00:00:00         1        NaN         1  PRJNA268708  ...   \n",
       "2     2019-01-01 00:00:00         1        NaN         1  PRJNA268708  ...   \n",
       "3     2019-01-01 00:00:00         1        NaN         1  PRJNA268708  ...   \n",
       "4     2019-01-01 00:00:00         1        NaN         1  PRJNA268708  ...   \n",
       "...                   ...       ...        ...       ...          ...  ...   \n",
       "2436  2019-06-05 10:02:31         1        NaN         1   PRJNA46339  ...   \n",
       "2437  2019-06-05 10:02:31         1        NaN         1   PRJNA46339  ...   \n",
       "2438  2019-06-05 10:02:31         1        NaN         1   PRJNA46339  ...   \n",
       "2439  2019-06-05 10:02:31         1        NaN         1   PRJNA46339  ...   \n",
       "2440  2019-06-05 10:02:31         1        NaN         1   PRJNA46339  ...   \n",
       "\n",
       "         sex host_age diet  longitude latitude BMI Recent.Antibiotics.Use  \\\n",
       "0        NaN      NaN  NaN     2.1833  41.3833 NaN                      N   \n",
       "1        NaN      NaN  NaN     2.1833  41.3833 NaN                      N   \n",
       "2        NaN      NaN  NaN     2.1833  41.3833 NaN                      N   \n",
       "3        NaN      NaN  NaN     2.1833  41.3833 NaN                      N   \n",
       "4        NaN      NaN  NaN     2.1833  41.3833 NaN                      N   \n",
       "...      ...      ...  ...        ...      ...  ..                    ...   \n",
       "2436    Male      NaN  NaN        NaN      NaN NaN                    NaN   \n",
       "2437    Male      NaN  NaN        NaN      NaN NaN                    NaN   \n",
       "2438  Female      NaN  NaN        NaN      NaN NaN                    NaN   \n",
       "2439    Male      NaN  NaN        NaN      NaN NaN                    NaN   \n",
       "2440    Male      NaN  NaN        NaN      NaN NaN                    NaN   \n",
       "\n",
       "     antibiotics_used  Antibiotics.Dose Days.Without.Antibiotics.Use  \n",
       "0                 NaN               NaN                          NaN  \n",
       "1                 NaN               NaN                          NaN  \n",
       "2                 NaN               NaN                          NaN  \n",
       "3                 NaN               NaN                          NaN  \n",
       "4                 NaN               NaN                          NaN  \n",
       "...               ...               ...                          ...  \n",
       "2436              NaN               NaN                          NaN  \n",
       "2437              NaN               NaN                          NaN  \n",
       "2438              NaN               NaN                          NaN  \n",
       "2439              NaN               NaN                          NaN  \n",
       "2440              NaN               NaN                          NaN  \n",
       "\n",
       "[2441 rows x 38 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_IBD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "project_id\n",
       "PRJEB11419     1956\n",
       "PRJNA268708     290\n",
       "PRJNA392762      91\n",
       "PRJNA386442      48\n",
       "PRJNA46339       48\n",
       "PRJNA302437       8\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_IBD['project_id'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "IBD_train_df = metadata_IBD[metadata_IBD['project_id'] == 'PRJEB11419']\n",
    "\n",
    "# Create the testing DataFrame where project_id is not 'PRJEB11419'\n",
    "IBD_test_df = metadata_IBD[metadata_IBD['project_id'] != 'PRJEB11419']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1956, 38)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "IBD_train_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(485, 38)"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "IBD_test_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "disease\n",
       "D006262    1376\n",
       "D043183     580\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "IBD_train_df['disease'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "disease\n",
       "D043183    374\n",
       "D006262    111\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "IBD_test_df['disease'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "sex\n",
       "Female    974\n",
       "Male      972\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "IBD_train_df['sex'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "sex\n",
       "Female    131\n",
       "Male       52\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "IBD_test_df['sex'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1956"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "IBD_train_df['host_age'].count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "135"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "IBD_test_df['host_age'].count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1956"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "IBD_train_df['BMI'].count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "IBD_test_df['BMI'].count()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "confounder_free",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
